c++/src/kj/compat/http.c++

```diff
@@ -4956,9 +5054,12 @@ public:
 
     auto id = ++counter;
 
-    return httpInput.readResponseHeaders().then(
-        [this,id](HttpHeaders::ResponseOrProtocolError&& responseOrProtocolError)
-            -> HttpClient::ConnectResponse {
+    auto guard = kj::newPromiseAndFulfiller<HttpInputStreamImpl::ReleasedBuffer>();
```

Comment:
It looks like if the caller drops the `status` promise without waiting for it, then the operation will be canceled, dropping this fulfiller, causing the promise to throw, so the stream will end up broken. I think waiting on `status` should be optional; if either `status` or `connection` is kept alive, the connection attempt should continue. So you'll need to `fork()` or `split()` a promise here instead of using PromiseAndFulfiller.